git-branchless/tests/test_amend.rs

```diff
@@ -412,6 +412,41 @@ fn test_amend_head() -> eyre::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_amend_head_with_file_with_space() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    if !git.supports_committer_date_is_author_date()? {
+        return Ok(());
+    }
+
+    git.init_repo()?;
+    git.detach_head()?;
+    git.commit_file("test file with space", 1)?;
+    git.write_file_txt("test file with space", "updated contents")?;
```

Comment:
I added a test for `git amend` because that's where I first encountered this, while working on some documentation.